std/conv.d

```diff
@@ -3926,6 +3926,8 @@ if (is(UT == Unqual!T))
     {
         static assert (is(typeof({static T i;})),
             convFormat("Cannot emplace a %1$s because %1$s.this() is annotated with @disable.", T.stringof));
+        static if (is(T == class)) static assert (!isAbstractClass!T,
+            T.stringof ~ " is abstract and it can't be emplaced");
```

Comment:
@JakobOvrum is right. The conditions in `static if` or `static assert` will be evaluated with short circuit behavior.
When a static assert condition is `A || B`, and `A` is evaluated `false`, `B` __will never be analyzed__ at compile time. So in here, you can use a bare static assert:

```d
static assert(!is(T == class) || !isAbstractClass!T, ...);
```